### PR TITLE
Fixed #34572 -- Added support for GDAL 3.7.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-34572

### CODE REVIEW

File: tests/gis_tests/gdal_tests/test_raster.py [hunk 1]
```diff
@@ -448,9 +458,12 @@ def test_compressed_file_based_raster_creation(self):
...
         )
-        self.assertEqual(
-            compressed.bands[0].metadata["IMAGE_STRUCTURE"]["PIXELTYPE"], "SIGNEDBYTE"
-        )
+        self.assertEqual(compressed.bands[0].datatype(), datatype)
+        if GDAL_VERSION < (3, 7):
+            self.assertEqual(
+                compressed.bands[0].metadata["IMAGE_STRUCTURE"]["PIXELTYPE"],
+                "SIGNEDBYTE",
+            )
```
Comment (felixxm) [inline tests/gis_tests/gdal_tests/test_raster.py:466]: Check out [docs](https://gdal.org/user/raster_data_model.html#image-structure-domain):
> 
> _"Starting with GDAL 3.7, this metadata item is no longer used, as the Int8 data type is used for signed bytes."_

File: docs/ref/contrib/gis/install/geolibs.txt [hunk 2]
```diff
@@ -35,6 +35,7 @@ totally fine with GeoDjango. Your mileage may vary.
     GDAL 3.4.0 2021-11-04
     GDAL 3.5.0 2022-05-13
     GDAL 3.6.0 2022-11-03
+    GDAL 3.7.0 2023-05-02
```
Comment (smithdc1) [inline docs/ref/contrib/gis/install/geolibs.txt:38]: Maybe May 10th?  🤷 

https://github.com/OSGeo/gdal/releases
https://gdal.org/download.html#current-release

File: tests/gis_tests/gdal_tests/test_raster.py [hunk 3]
```diff
@@ -415,9 +415,24 @@ def test_compressed_file_based_raster_creation(self):
...
+                "blockxsize": 23,
+                "blockysize": 23,
+            }
+        else:
+            datatype = 14
+            papsz_options = {
+                "compress": "packbits",
+                "blockxsize": 23,
+                "blockysize": 23,
+            }
```
Comment (smithdc1) [inline tests/gis_tests/gdal_tests/test_raster.py:432]: Maybe 🤷 

```suggestion
        papsz_options = {
            "compress": "packbits",
            "blockxsize": 23,
            "blockysize": 23,
        }
        if GDAL_VERSION < (3, 7):
            datatype = 1
            papsz_options["pixeltype"] = "signedbyte"
        else:
            datatype = 14
```

**Comment by smithdc1** [comment]
I see one failing test, but I think it's not related to this change. I think it's related to https://github.com/django/django/commit/1b754d638dc5682265b5b4d1fcf5a94d0312113e


```
======================================================================
FAIL: test_centroid_collect_filter (gis_tests.relatedapp.tests.RelatedGeoModelTest.test_centroid_collect_filter)
----------------------------------------------------------------------
AssertionError: 'POINT (3.2127999999999997 1.5000000000000002)' != 'POINT (3.2128 1.5)'
- POINT (3.2127999999999997 1.5000000000000002)
+ POINT (3.2128 1.5)
```

I tested with:

```
proj==9.2.0
gdal==3.7.0
geos==3.11.2
libspatialite==5.0.1
```

**Review by smithdc1 — APPROVED** [review]
LGTM -- couple small questions below.

File: docs/ref/contrib/gis/install/geolibs.txt (hunk 2 above)
Comment (felixxm) [reply docs/ref/contrib/gis/install/geolibs.txt:38]: Changed :+1:

File: tests/gis_tests/gdal_tests/test_raster.py (hunk 3 above)
Comment (felixxm) [reply tests/gis_tests/gdal_tests/test_raster.py:432]: Changed :+1:

**Comment by felixxm** [comment]
@smithdc1 Thanks for the review :star: 

> I see one failing test, but I think it's not related to this change. I think it's related to [1b754d6](https://github.com/django/django/commit/1b754d638dc5682265b5b4d1fcf5a94d0312113e)

Yes, it's not related I have the same error with GDAL 3.5 and 3.6, will fix it separately.

**Comment by felixxm** [comment]
> Yes, it's not related I have the same error with GDAL 3.5 and 3.6, will fix it separately.

- #17031

**Pull request merged**